recipes/catch2/3.x.x/conanfile.py

```diff
@@ -0,0 +1,132 @@
+import os
+import functools
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+
+required_conan_version = ">=1.43.0"
+
+class ConanRecipe(ConanFile):
+    name = "catch2"
+    description = "A modern, C++-native, header-only, framework for unit-tests, TDD and BDD"
+    topics = ("conan", "catch2", "header-only", "unit-test", "tdd", "bdd")
+    license = "BSL-1.0"
+    homepage = "https://github.com/catchorg/Catch2"
+    url = "https://github.com/conan-io/conan-center-index"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "fPIC": [True, False],
+        "with_prefix": [True, False],
+        "default_reporter": "ANY",
+    }
+    default_options = {
+        "fPIC": True,
+        "with_prefix": False,
+        "default_reporter": None,
+    }
+    generators = "cmake"
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    @property
+    def _default_reporter_str(self):
+        return '"{}"'.format(str(self.options.default_reporter).strip('"'))
+
+    def export_sources(self):
+        self.copy("CMakeLists.txt")
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            self.copy(patch["patch_file"])
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "gcc": "7",
+            "Visual Studio": "15",
+            "clang": "5",
+            "apple-clang": "10",
+        }
+
+    def validate(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            tools.check_min_cppstd(self, "14")
+        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+        if minimum_version:
+            if tools.Version(self.settings.compiler.version) < minimum_version:
+                raise ConanInvalidConfiguration("structopt: Unsupported compiler: {}-{} "
```

Comment:
It seems to be copy-pasted from `structopt` recipe. Should be fixed. 

But I'm not sure that the current `_compiles_minimum_version` is correct. I didn't see this information in Catch2 repo, maybe we need to ask maintainer about it. Or just remove `_compilers_minimum_version` part from this recipe.

Reply:
> It seems to be copy-pasted from structopt recipe. Should be fixed.
Sorry. I copy validate() functions from `structopt`.
I fixed.

> I didn't see this information in Catch2 repo, maybe we need to ask maintainer about it. 

Catch2/3.x.x requires C++14.

To follow the following FAQ, I created compiler_minimum_version.
https://github.com/conan-io/conan-center-index/blob/master/docs/faqs.md#why-is-a-toolscheck_min_cppstd-call-not-enough

I know it is not public supported lists.
It must be updated later if there are more supported versions.